### DISCUSSION TITLE
State doesn't get passed on to form component

### DISCUSSION THREAD

**Discussion by StefanieLoeser** [neuefische/questions#231] [Web] [answered]
Hi everybody,

I tried to set up a form-component that gets called within my RecordFile component to enter a YouTube-URL. It was created with useForm from react-hook-form that is supposed to retrieve the video-id from the entered data (I found a react library that does exactly that). I then want to update my existing collection data stored with the video-ids. It would only be one global state on the _app.js level that gets passed down through the components and filtered for the specific needs. But when I try to log out the state that I received as props inside the form-component it is undefined. 

[Link to PR](https://github.com/StefanieLoeser/capstone-project/pull/20)

I'm not entering the code as it seems like it would be too much and it probably only makes sense to review within the PR. 

This is the error or result of my console.logs though. I checked if the id of the record where the form was clicked gets tracked, it does, but the collection in total is undefined. 

<img width="465" alt="Bildschirmfoto 2022-10-09 um 23 43 09" src="https://user-images.githubusercontent.com/104068626/194780783-dc775619-8222-4c5b-8a13-acb02944751c.png">

Many thanks!
Stefanie

**Comment by StefanieLoeser** [comment]
An additional note: I tried to to make handleSubmit an anonymous function call as I try to pass parameters, but it doesn't work as the  syntax `onSubmit={handleSubmit(onSubmit)}` is specific for the react-hook-form library. I tried to research wether it is even possible to pass parameters but can't really find an answer that seems clear to me. Also logging out the recordID shows that the info does land in the function (as shows the screenshot above).

**Comment by ThomasKirst** [comment, marked as answer]
Hi Stefanie,
I don´t really understand why you use a form library since you have only one input field.
But you could try to pass your parameters most probably like this:
```js
<form
  onSubmit={handleSubmit((data) => onSubmit(data, recordID, collection, onSetCollection))}
>
```

EDIT: make also sure to pass all required props into the form component:

```jsx
<YouTubeURLForm recordID={record.id} />
```
–> `collection` and `onSetCollection()` are missing.

Does this help?

**Reply by StefanieLoeser** [reply]
Thank you, @ThomasKirst. 

Unfortunately both suggestions didn't help. 

I had tried to pass the state as props this way before but that didn't change anything. Then I thought as the state is set within _app.js I should be able to receive it directly from there and I only need the ID being passed through. 

I'll try to rebuild the form without the library unless you have any other idea?

**Reply by ThomasKirst** [reply]
Is there still anything open or can the discussion be closed?

**Reply by ThomasKirst** [reply]
@StefanieLoeser 👆 Solved？

**Reply by StefanieLoeser** [reply]
Yes, thank you! 🙏
